### PR TITLE
Use approximate queue length for azure queue scaler

### PR DESCRIPTION
**Pull request by moria97**
Use the approximate queue length instead of visible messages, see #4002 
### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/4002

Relates to #

### CODE REVIEW

**Comment by tomkerkhove** [comment]
By removing the visible count you are not actually achieving what you are suggesting in https://github.com/kedacore/keda/issues/4002

**Comment by moria97** [comment]
> By removing the visible count you are not actually achieving what you are suggesting in #4002

Removing the visible count will return the approximate message count directly, which contains messages that are queued and being processed. Not sure if there is anything wrong? @tomkerkhove

**Review by v-shenoy — APPROVED** [review]
LGTM

**Comment by tomkerkhove** [comment]
I have misread the issue, no worries.

**Comment by tomkerkhove** [comment]
Can you add an entry in the changelog please?

I've pinged @ahmelsayed privately as well to see if we can remember why we added that.

**Comment by tomkerkhove** [comment]
@ahmelsayed pointed out that this was changed in https://github.com/kedacore/keda/pull/2613 by @RamCohen as an optimization.

I think the optimization is still OK to have and we can skip this PR but I don't have a strong opinion.

**Comment by moria97** [comment]
@tomkerkhove  This optimization looks make sense to me. But the issue is not related to the optimization, but the visible message count used to count queue length.

 Let's say "queue length" parameter is 1, there are 20 messages and scaled out to 20 instances and pulled message from queue, the queue size returned is 0 （visible message account）, but the actual value should be 20 (approximate queue length).

**Review by zroubalik — COMMENTED** [review]
@moria97 @tomkerkhove @v-shenoy so what is the status of this PR?

**Comment by antonio-policastro** [comment]
I'm shipping a custom version of this scaler in production since v2.6.1. My custom versions have the exact same code as in this pull request, and they have been working perfectly fine for the past year. ApproximateMessageCount is the way to go.
There are multiple issues about this problem(eg: #2385 #3555).
I hope this will be merged soon.
Thanks

**Comment by moria97** [comment]
@tomkerkhove @zroubalik Can you take a look at this pr?

**Review by v-shenoy — APPROVED** [review]
I approved this earlier also. I think we can (should) merge this. 

@tomkerkhove @JorTurFer

File: CHANGELOG.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ Here is an overview of all new **experimental** features:
 - **Kafka Scaler**: Support 0 in activationLagThreshold configuration ([#4137](https://github.com/kedacore/keda/issues/4137))
 - **Prometheus Metrics**: Expose Prometheus Metrics also when getting ScaledObject state ([#4075](https://github.com/kedacore/keda/issues/4075))
 - **Redis Scalers**: Fix panic produced by incorrect logger initialization ([#4197](https://github.com/kedacore/keda/issues/4197))
+- **Azure Queue Scaler** Fix azure queue length ([#4002](https://github.com/kedacore/keda/issues/4002))
```
Comment (JorTurFer) [inline CHANGELOG.md:90]: Could you move this before **Azure Service Bus Scaler:**? We shot the changelog, putting `General` on top and the alphabetically shorted, **Azure Queue Scaler** goes before **Azure Service Bus Scaler:**

**Review by JorTurFer — APPROVED** [review]
LGTM

**Comment by tomkerkhove** [comment]
> @ahmelsayed pointed out that this was changed in #2613 by @RamCohen as an optimization.
> 
> I think the optimization is still OK to have and we can skip this PR but I don't have a strong opinion.

I still think this optimization does not hurt though. Can you elaborate on why you think it's bad @moria97?

File: CHANGELOG.md (hunk 1 above)
Comment (moria97) [reply CHANGELOG.md]: Updated. Thanks!

**Comment by wallflower762** [comment]
Thanks team, this is great!

**Pull request merged**